packages/astro-embed-bluesky/src/post.astro

```diff
@@ -0,0 +1,157 @@
+---
+import { atUriToPostUri, renderPostAsHtml, resolvePost } from './utils.js';
+import Embed from './embed.astro';
+import type { Post } from './types';
+import './styles.css';
+import Avatar from './avatar.astro';
+
+type Props = {
+	id?: string;
+	post?: string | Post;
+};
+
+const postRef = Astro.props.id ?? Astro.props.post;
+
+if (!postRef) {
+	return new Response('');
+}
+
+const post = await resolvePost(postRef)!;
+
+if (!post) {
+	return new Response('');
+}
+
+const postUrl = atUriToPostUri(post.uri);
+
+const { record, embed, author } = post;
+
+const authorUrl = `https://bsky.app/profile/${author?.handle}`;
+
+const body = renderPostAsHtml(post);
+
+const formatter = new Intl.DateTimeFormat('en-US', {
+	month: 'long',
+	day: 'numeric',
+	year: 'numeric',
+	hour: 'numeric',
+	minute: 'numeric',
+	timeZone: 'UTC',
+	timeZoneName: 'short',
+});
+---
+
+<div class="bluesky-post-container not-content">
+	<div class="post-content">
+		<div class="post-header">
+			<Avatar user={author} link={true} />
+			<div class="user-info">
+				<a href={authorUrl} class="display-name">{author?.displayName}</a>
+				<a href={authorUrl} class="username">@{author?.handle}</a>
+			</div>
+			<a href={postUrl} class="logo-link">
+				<img
+					src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20fill='none'%20viewBox='0%200%20320%20286'%3e%3cpath%20fill='rgb(10,122,255)'%20d='M69.364%2019.146c36.687%2027.806%2076.147%2084.186%2090.636%20114.439%2014.489-30.253%2053.948-86.633%2090.636-114.439C277.107-.917%20320-16.44%20320%2032.957c0%209.865-5.603%2082.875-8.889%2094.729-11.423%2041.208-53.045%2051.719-90.071%2045.357%2064.719%2011.12%2081.182%2047.953%2045.627%2084.785-80%2082.874-106.667-44.333-106.667-44.333s-26.667%20127.207-106.667%2044.333c-35.555-36.832-19.092-73.665%2045.627-84.785-37.026%206.362-78.648-4.149-90.071-45.357C5.603%20115.832%200%2042.822%200%2032.957%200-16.44%2042.893-.917%2069.364%2019.147Z'/%3e%3c/svg%3e"
+					class="bluesky-logo"
+					alt="Bluesky"
+				/>
+			</a>
+		</div>
+
+		<p class="post-text" set:html={body} />
+
+		{embed && <Embed embed={embed} postUrl={postUrl} />}
+		<a href={postUrl} class="timestamp">
+			{formatter.format(new Date(record.createdAt ?? ''))}
```

Comment:
Added some tests which failed in CI because of a different timezone compared to running locally which made me think again about this. I now remember we hit the same question with the old Twitter embed — if we render a timestamp on the server and therefore in the server’s time zone, that can be confusing because it is not local to the user (unlike the client-rendered components).

I think for Twitter I just decided to display a date with no time IIRC (still has a chance of being off, but it’s not as likely).

There’s also the option of some minimal JS hydration, e.g. to render something like this which updates the time to the user timezone:

```astro
<bluesky-time>
  <time datetime={record.createdAt?.toISOString()}>
    {formatter.format(new Date(record.createdAt ?? ''))}
  </time>
</bluesky-time>

<script>
const formatter = new Intl.DateTimeFormat('en-US', { dateStyle: 'long', timeStyle: 'short' });

customElements.define('bluesky-time', class BlueskyTime extends HTMLElement {
	connectedCallback() {
		try {
			const time = this.querySelector('time');
			const datetime = time?.getAttribute('datetime');
			if (!datetime) return;
			const date = new Date(datetime);
			time.textContent = formatter.format(date);
		} catch {}
	}
});
</script>
```

Does away with the zero-JS claim though.

Reply:
NEVER